### ISSUE TITLE
Use ES6 modules

### ISSUE THREAD

**Issue by sanfilippopablo** [Turfjs/turf#790]
Hello. Awesome project. Is there a possibility of exporting ES6 modules? This would allow to install just the main turf dependency and then let tree shaking remove everything unused.

**Comment by erikh2000** [comment]
Could you get the same benefit by using an ES6 import statement declaring specific functions your modules will use? E.g. import { difference } from '@turf/turf'

**Comment by sanfilippopablo** [comment]
Full disclosure: I'm saying this by reading the source of the turf package, I haven't actually tested it.

So, currently the docs recommend to install each package corresponding to the function you use. For example, `npm install @turf/sector` and then `import sector from '@turf/sector'`. if you use multiple packages, you have to chose either:
* Install every package you use separately and require it like that.
* Install `@turf/turf` and the do exactly what you say: `import { difference } from '@turf/turf'`.

The problem with that second option is that the bundler will include in the bundler *every* function, because the turf module includes every function, even if you don't import it in any module. That's like this because the turf package index.js [requires every function](https://github.com/Turfjs/turf/blob/master/packages/turf/index.js).

However, if that file instead of exporting like this:
```
isoline: require('@turf/isoline')
```
exports like this:
```
import isoline from '@turf/isoline'

export isoline
```
the exported modules would be ES6 modules. That means that bundlers that support tree shaking (webpack and rollup) could remove unused code, even if you install the main turf package.

**Comment by erikh2000** [comment]
Good explanation, thank you. I was hoping webpack tree-shaking would figure out the minimal dependencies with the current 'require' declarations. But I guess it doesn't from what you say. I'm no expert on webpack, myself.

**Comment by DenisCarriere** [comment]
We could use ES6 modules **ONLY** in `packages/turf`, we're already doing this in the Typescript definition ([turf/index.d.ts](https://github.com/Turfjs/turf/blob/master/packages/turf/index.d.ts)). 

As for all the TurfJS modules, they should remain in ES5 since NodeJS still does not natively support `import` & `export` and we should continue to support NodeJS 4 as a minimum (without any compiler).

I'm 👍 if `@turf/turf` would have two entry points in the `package.json`:

- [x] `"main": "index.js"` => Browserify build
- [ ] `"module": "module.js"` => ES6 build using Typescript as the compiler

CC: @morganherlocker

**Comment by sanfilippopablo** [comment]
Exactly. That's at least what it says [in the Rollup docs](https://github.com/rollup/rollup/wiki/pkg.module). I thought webpack also supported it, but there's an [issue](https://github.com/webpack/webpack/issues/1979) with a back and forth about this. Give me a day and I'll test that pull request you created against rollup and webpack to see if it works.

**Comment by DenisCarriere** [comment]
👍 Sounds like a plan.

**Comment by sanfilippopablo** [comment]
I'm sorry, I've been really busy, and didn't have the time to do it. Hopefully this week I'll test it.

**Comment by DenisCarriere** [comment]
No worries, adding 'module' sounds like the best solution.

**Comment by stebogit** [comment]
Implemented with PR #793.
@sanfilippopablo feel free to open this or a new issue if your tests suggest so.

**Comment by allthesignals** [comment]
I'm confused - I can't seem to import bbox with `import { bbox } from '@turf/turf';`. How do I import as an ES6 module? 

`Could not find module `@turf/turf``
